### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by Diarqq**

### CODE REVIEW

File: 13.1.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+        while j < 3:
+            yield j
+            j+=1
+            i+=1
+        else:
+            j = 0
+limit = int(input("Set the limit - "))
+r = custom_range(limit)
+for i in range(limit):
+    print(next(r))
```
Comment (SergeySmirnov1993) [inline 13.1.py:14]: we should add blank lines at the end of the file

File: 13.1.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+def custom_range(limit):
+    i=0
+    j=0
+    while i < limit:
```
Comment (SergeySmirnov1993) [inline 13.1.py:4]: we do not need limit
the sequence should be infinite, in your case it is limited by stop

File: 13.1.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+def custom_range(limit):
```
Comment (SergeySmirnov1993) [inline 13.1.py:1]: we should accept sequence as a parameter

File: 13.2.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
+class Pizza:
+    def __init__(self):
+        self.size = None
+        self.cheese = None
+        self.pepperoni = None
+        self.mushrooms = None
+        self.onions = None
+        self.bacon = None
```
Comment (SergeySmirnov1993) [inline 13.2.py:8]:
```suggestion
   def __init__(self, size=None, cheese=False, pepperoni=False,
                 mushrooms=False, onions=False, bacon=False):
        self.size = size
        self.cheese = cheese
        self.pepperoni = pepperoni
        self.mushrooms = mushrooms
        self.onions = onions
        self.bacon = bacon
```

File: 13.2.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        print( f'Onions -  {onions}')
+
+    def choose_bacon(self,bacon):
+        self.pizza.bacon = bacon
+        print( f'Bacon - {bacon}')
+
+
+class PizzaDirector:
+    def __init__(self):
+        self.builder = PizzaBuilder()
```
Comment (SergeySmirnov1993) [inline 13.2.py:42]:
```suggestion
    def __init__(self, builder):
        self.builder = builder

```

File: 13.3.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
+from abc import ABC, abstractmethod
+
```
Comment (SergeySmirnov1993) [inline 13.3.py:2]: after imports should be 2 blank lines

File: 13.3.py [hunk 7]
```diff
@@ -0,0 +1,35 @@
+from abc import ABC, abstractmethod
+
+class Animal(ABC):
+    def __init__(self,type):
+        self.type = type
+
+    @abstractmethod
+    def speak(self):
+        pass
+
```
Comment (SergeySmirnov1993) [inline 13.3.py:10]: after class definitions should be 2 blank lines

File: 13.3.py [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+class Cat(Animal):
+    def __init__(self):
+        super().__init__("Cat")
+
+    def speak(self):
+        print("Meoow")
+
+class AnimalFactory:
+    def create_animal(animal_type):
```
Comment (SergeySmirnov1993) [inline 13.3.py:26]: we can do this as a static method

File: 13.3.py [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+        if animal_type.lower() == "dog":
+            return Dog()
+        elif animal_type.lower() == "cat":
+            return Cat()
+        else:
+            raise ValueError("Unknown animal type")
+
+dog = AnimalFactory.create_animal('dog')
+dog.speak()
```
Comment (SergeySmirnov1993) [inline 13.3.py:35]: add please blank line at the end of the file

**Review by SergeySmirnov1993 — APPROVED** [review]
Please rename the PR

**Review by Diarqq — COMMENTED** [review]
everything is updated

**Pull request merged**